mono/utils/mono-threads.h

```diff
@@ -532,6 +532,13 @@ mono_native_thread_create (MonoNativeThreadId *tid, gpointer func, gpointer arg)
 MONO_API void
 mono_native_thread_set_name (MonoNativeThreadId tid, const char *name);
 
+#ifdef HOST_WIN32
+void
+mono_native_thread_set_namew (HANDLE threadHandle, const wchar_t *name);
```

Comment:
It seems odd that `mono_native_thread_set_name` would take an id while `mono_native_thread_set_namew ` would take a handle.

Reply:
Yes, the old Visual Studio-oriented code is indeed wierd.
RaiseException? Not usable from standard C?
The new code is what you would expect.

In the face of wierdness, is it better to be consistent and wierd
or non-wierd?

I am the implementer of SetThreadDescription, and it being handle-based
is no doubt how it should be, not wierd.

Yes, OpenThread would work. I had actually coded that up, before
remembering that mono has the proper handle. But it is wierd and less
efficient.